mne/io/nsx/tests/test_nsx.py

```diff
@@ -0,0 +1,183 @@
+    assert times.shape[0] == n_times
+    # Check data
+    # There is only one contiguous data packet, samples 0--100. Data
+    # was generated as:
+    #     ```data = np.ones((n_samples, ch_count))
+    #        data[n_samples // 2] = np.arange(ch_count) + 10
+    #        data[:, ch_count // 2] = np.arange(n_samples) + 100```
+    orig_data = raw_data / (raw.info["chs"][0]["cal"] * raw.info["chs"][0]["range"])
+    assert np.allclose(sum(orig_data[:, 50] - 10 - np.arange(n_channels)), 76.0)
+
+    assert np.allclose(orig_data[n_channels // 2, :100] - 100, np.arange(100))
+
+    data, times = raw.get_data(start=10, stop=20, return_times=True)
+    assert n_channels, 10 == data.shape
+
+    data, times = raw.get_data(start=0, stop=300, return_times=True)
+    epochs = make_fixed_length_epochs(raw, duration=0.05, preload=True, id=1)
+    assert len(epochs) == 1
+    assert epochs.event_id["1"] == 1
+    with pytest.raises(ValueError) as excinfo:
+        _ = make_fixed_length_epochs(raw, duration=0.5, preload=True)
+    assert "No events produced, check the values of start, stop, and duration" in str(
+        excinfo.value
+    )
+
+
+@requires_testing_data
+def test_stim_eog_misc_chs_in_nsx():
+    """Test stim/misc/eog channel assignments."""
+    raw = read_raw_nsx(nsx_22_fname, stim_channel="elec127", eog=["elec126"])
+    assert raw.info["chs"][127]["kind"] == FIFF.FIFFV_STIM_CH
+    assert raw.info["chs"][126]["kind"] == FIFF.FIFFV_EOG_CH
+    raw = read_raw_nsx(nsx_22_fname, stim_channel=["elec127"], eog=["elec126"])
+    assert raw.info["chs"][127]["kind"] == FIFF.FIFFV_STIM_CH
+    assert raw.info["chs"][126]["kind"] == FIFF.FIFFV_EOG_CH
+    raw = read_raw_nsx(nsx_22_fname, stim_channel=127, eog=["elec126"])
+    assert raw.info["chs"][127]["kind"] == FIFF.FIFFV_STIM_CH
+    assert raw.info["chs"][126]["kind"] == FIFF.FIFFV_EOG_CH
+    raw = read_raw_nsx(nsx_22_fname, stim_channel=[127], eog=["elec126"])
+    assert raw.info["chs"][127]["kind"] == FIFF.FIFFV_STIM_CH
+    assert raw.info["chs"][126]["kind"] == FIFF.FIFFV_EOG_CH
+    stims = [ch_info["kind"] == FIFF.FIFFV_STIM_CH for ch_info in raw.info["chs"]]
+    assert np.any(stims)
+    assert raw.info["chs"][126]["kind"] == FIFF.FIFFV_EOG_CH
+    with pytest.raises(ValueError) as excinfo:
+        raw = read_raw_nsx(nsx_22_fname, stim_channel=["elec128", 129], eog=["elec126"])
+    assert "Invalid stim_channel" in str(excinfo.value)
+    with pytest.raises(ValueError) as excinfo:
+        raw = read_raw_nsx(nsx_22_fname, stim_channel=("elec128",), eog=["elec126"])
+    assert "Invalid stim_channel" in str(excinfo.value)
+
+    raw = read_raw_nsx(nsx_22_fname, stim_channel="elec127", misc=["elec126", "elec1"])
+    assert raw.info["chs"][126]["kind"] == FIFF.FIFFV_MISC_CH
+    assert raw.info["chs"][1]["kind"] == FIFF.FIFFV_MISC_CH
+
+
+@requires_testing_data
+def test_nsx_ver_21():
+    """Primary tests for NEURALSG reader."""
+    with pytest.raises(NotImplementedError):
```

Comment:
Every `with pytest.raises` context should have a `match` arg for two reasons: 1) makes test more targeted, 2) makes it clearer what the check actually checks. I'd expect something like this but please adjust to the actual message:
```suggestion
    with pytest.raises(NotImplementedError, match=".*format 21.*not supported.*"):
```